File: src/app/RioValleyChili.Core/Models/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using RioValleyChili.Core.Helpers;

namespace RioValleyChili.Core.Models
{
    [ComplexType]
    public class Address
    {
        [StringLength(Constants.StringLengths.AddressLine)]
        [Display(Name = "Address Line 1")]
        public string AddressLine1 { get; set; }

        [StringLength(Constants.StringLengths.AddressLine)]
        [Display(Name = "Address Line 2")]
        public string AddressLine2 { get; set; }

        [StringLength(Constants.StringLengths.AddressLine)]
        [Display(Name = "Address Line 3")]
        public string AddressLine3 { get; set; }

        [StringLength(Constants.StringLengths.AddressCity)]
        public string City { get; set; }

        [StringLength(Constants.StringLengths.AddressState)]
        public string State { get; set; }

        [StringLength(Constants.StringLengths.AddressPostalCode)]
        [Display(Name = "Postal Code")]
        [DataType(DataType.PostalCode)]
        public string PostalCode { get; set; }

        [StringLength(Constants.StringLengths.AddressCountry)]
        public string Country { get; set; }
    }
}

File: src/app/RioValleyChili.Client.Mvc/Scripts/app/viewModels/shared/address.js
function Address(data) {
    data = ko.utils.unwrapObservable(data) || { };
    var me = this;
    me.AddressLine1 = ko.observable(data.AddressLine1);
    me.AddressLine2 = ko.observable(data.AddressLine2);
    me.AddressLine3 = ko.observable(data.AddressLine3);
    me.City = ko.observable(data.City);
    me.State = ko.observable(data.State);
    me.PostalCode = ko.observable(data.PostalCode);
    me.Country = ko.observable(data.Country);

    return me;
}